shapely/strtree.py

```diff
@@ -342,28 +314,55 @@ def nearest(self, geom, exclusive: bool = False) -> Union[Any, None]:
         if self._tree.count == 0:
             return None
 
-        return self._nearest_idx(geom, exclusive)
+        geometry_arr = np.asarray(geometry, dtype=object)
+        # TODO those changes compared to _tree.nearest output should be pushed into C
```

Comment:
I'm unclear on the meaning of this comment (it came originally from `_nearest_idx` which is replaced here).  Is this something we should tackle in this PR too?

Reply:
I think that I wrote this, and that it meant that we can in principle let the C tree code handle this check. Because we currently have this in STRtree_nearest:

https://github.com/shapely/shapely/blob/9b3fa0ae3dfa802a56bb01b21ccf8411a20d5a9d/src/strtree.c#L680-L682

which has different behaviour (it doesn't raise an error but actually ignore missing or empty geometries). Although in practice we should also never end up in that if check now, given that we check here in advance for it (and keeping the check in python might be easier to raise a custom error).

Reply:
Doing it up front is maybe a little bit of extra overhead (up to 2 passes over the array) compared to moving into C as part of our main loop, but it provides the advantage of raising the error earlier rather than partway through finding nearest neighbors.

I think we can delete the comment?

Reply:
> I think we can delete the comment?

Yes, sounds good